src/nvim/ex_cmds2.c

```diff
@@ -2635,6 +2635,59 @@ static void cmd_source(char_u *fname, exarg_T *eap)
   }
 }
 
+/// Concatenate VimL line if it starts with a line continuation into a growarray
+/// (excluding the continuation chars and leading whitespace)
+///
+/// @note Growsize of the growarray may be changed to speed up concatenations!
+///
+/// @param ga  the growarray to append to
+/// @param init_growsize  the starting growsize value of the growarray
+/// @param p  pointer to the beginning of the line to consider
+/// @param len  the length of this line
+///
+/// @return true if this line did begin with a continuation; the next line
+///         should also be considered, if any
+static bool concat_continued_line(garray_T *const ga, const int init_growsize,
+                                  const char_u *const p, size_t len)
+  FUNC_ATTR_NONNULL_ALL
+{
+  const char_u *const line = skipwhite_len(p, len);
+  len -= (size_t)(line - p);
+  // Skip lines starting with '\" ', concat lines starting with '\'
+  if (len >= 3 && STRNCMP(line, "\"\\ ", 3) == 0) {
+    return true;
+  } else if (len == 0 || line[0] != '\\') {
+    return false;
+  }
+  if (ga->ga_len > init_growsize) {
+    ga_set_growsize(ga, MAX(ga->ga_len, 8000));
+  }
+  ga_concat_len(ga, (const char *)line + 1, len - 1);
+  return true;
+}
+
+/// Concatenate (possibly many) VimL lines starting with line continuations into
+/// a growarray. @see concat_continued_line
+///
+/// @note All parameters, excluding `ga`, accept expressions that are evaluated
+///       once for each line; side-effects may be triggered many times!
+///
+/// @param ga  the growarray to append to
+/// @param cond  should evaluate to true if a next line exists
+/// @param line  should evaluate to the current line
+/// @param next  should handle fetching the next line when evaluated
+#define CONCAT_CONTINUED_LINES(ga, cond, line, next) \
+  do { \
+    garray_T *const ga_ = (ga); \
+    const int init_growsize_ = ga_->ga_growsize; \
+    for (; (cond); (next)) { \
+      const char_u *const line_ = (line); \
+      if (!concat_continued_line(ga_, init_growsize_, line_, STRLEN(line_))) { \
+        break; \
+      } \
+    } \
+  } while (false)
```

Comment:
I don't know what is the recomendation here, but I rather to this:
```suggestion
  { \
    garray_T *const ga_ = (ga); \
    const int init_growsize_ = ga_->ga_growsize; \
    for (; (cond); (next)) { \
      const char_u *const line_ = (line); \
      if (!concat_continued_line(ga_, init_growsize_, line_, STRLEN(line_))) { \
        break; \
      } \
    } \
  }
```

Reply:
By using `do {} while(false)` a `;` is required after the macro invocation, which I think is why it's used (apart from being able to `break` out of it)?